ddapm_test_agent/trace.py

```diff
@@ -275,11 +276,50 @@ def set_metric_tag(s: Span, k: str, v: MetricType) -> Span:
     return s
 
 
+def parse_json(json_string: bytes) -> Dict[str, Any]:
+    def is_number_as_str(num, number_type=int):
+        try:
+            number_type(num)
+            return isinstance(num, str)
+        except ValueError:
+            return False
+
+    # Define a custom JSON decoder for decoding spans
+    def json_decoder(maybe_span):
+        # loop through the span object
+        if isinstance(maybe_span, dict):
+            for key, value in maybe_span.items():
+                if key == "meta":
+                    # Check if the value is an int or float and convert back to string if true
+                    for k, v in value.items():
+                        if isinstance(v, int) or isinstance(v, float):
+                            value[k] = str(v)
+                elif key == "metrics":
+                    for k, v in value.items():
+                        # Check if value is a a float or int
+                        if is_number_as_str(v, int):
+                            value[k] = int(v)
+                        elif is_number_as_str(v, float):
+                            value[k] = float(v)
+                # For other attributes, check if the value is a string that can be converted to an int
+                elif is_number_as_str(value, int):
+                    maybe_span[key] = int(value)
+        return maybe_span
+
+    parsed_data: Dict[str, Any] = json.loads(json_string, object_hook=json_decoder)
+    return parsed_data
+
+
+_default_parse_function = (
```

Comment:
can we make this a runtime check instead of doing it at import-time?

Reply:
this would enable some extra flexibility in the future in-case we only want to do this per-test-case

Reply:
.... i think I addressed this?